src/quickpicks/creation.ts

```diff
@@ -1,8 +1,70 @@
+import * as semver from 'semver';
+import Appc from '../appc';
 import { nameForPlatform, platforms } from '../utils';
 import { CustomQuickPick, quickPick } from './common';
+import { appc } from 'titanium-editor-commons/updates';
+import { promisify } from 'util';
+
+export interface CodeBase {
+	android?: 'java' | 'kotlin'
+	ios?: 'objc' | 'swift'
+}
+
+interface AndroidCodeBaseQuickPickItem extends CustomQuickPick {
+	id: 'java' | 'kotlin'
+}
+
+interface iOSCodeBaseQuickPickItem extends CustomQuickPick {
+	id: 'objc' | 'swift'
+}
 
 export async function selectPlatforms (): Promise<string[]> {
 	const choices: CustomQuickPick[] = platforms().map(platform => ({ label: nameForPlatform(platform), id: platform, picked: true }));
 	const selected = await quickPick(choices, { canPickMany: true, placeHolder: 'Choose platforms' });
 	return selected.map((platform: CustomQuickPick)  => platform.id);
 }
+
+export async function selectCodeBases(platforms: string[]): Promise<CodeBase|undefined> {
+	const codeBases: CodeBase = {
+		android: undefined,
+		ios: undefined
+	};
+
+	if (!Appc.info) {
+		await promisify(Appc.getInfo).bind(Appc)();
```

Comment:
`getInfo` is currently callback based this is kind of a stopgap until we refactor it to be async/await based. I didn't do that in this PR as it would touch a few different places